### PR TITLE
Complete rewrite of Getting Started guide

### PR DESCRIPTION
**Pull request by jakelishman**
### Summary

This PR is the culmination of the documentation / on-boarding working
session from the Qiskit Dev conference in Tarrytown, April 2022.

The flagship changes are:

- the contributing guide is completely new, and targetted specifically
  at onboarding new external contributors, and helping them find issues
  to work on.  This includes a flowchart to direct people to
  repositories all across the Qiskit organisation, made by Abby and
  Anne.

- the old contributing guide is split up into separate pages, such as
  "building from source" and a "maintainer guide", so the information
  isn't quite so dense, and can be dipped into more easily.

- the deprecation policy is reorganised to make the processes more
  clear, and then later providing more details on how certain types of
  deprecation can be done.

### Details and comments

This makes the Getting Started guide the single (only) source of truth
for installing the metapackage components, moving the correct
information from the contributing guide to the getting started pages,
which were previously out-of-date.

This also splits the installation components of the getting started
pages into separate pages rather than separate tabs, because the
`..  tabbed::` directive didn't play nicely with Sphinx headings and
references, which caused all links to the source-installation sections
to be broken.

This also moves the deprecation policy into a separate file, and
rewrites the text to be shorter and more split it.  Also add sections on
the difficulty of changing behaviours and how to potentially avoid those
situations, and how to test the deprecated functionality correctly.

The actual policy should be unchanged by this commit.

Co-authored-by: Luciano Bello <bel@zurich.ibm.com>
Co-authored-by: Abby Mitchell <23662430+javabster@users.noreply.github.com>

This PR isn't meant to have actually changed any of our policies anywhere, just to reformat the all the documentation into more easily digestible forms.

### CODE REVIEW

File: docs/getting_started/local.rst [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+.. code:: sh
+
+    pip install qiskit[visualization]
+
+It is worth pointing out that if you're a zsh user (which is the default shell on newer
+versions of macOS), you'll need to put ``qiskit[visualization]`` in quotes:
+
+.. code:: sh
+
+    pip install 'qiskit[visualization]'
```
Comment (HuangJunye) [inline docs/getting_started/local.rst]: I know this part isn't new but I wonder whether we should simply suggest all users to use `pip install 'qiskit[visualization]'` without the need to mention the particularity of zsh.

File: docs/getting_started.rst [hunk 2]
```diff
@@ -641,8 +108,3 @@ Ready to get going?...
 .. raw:: html
 
    </div>
-
-.. Hiding - Indices and tables
-   :ref:`genindex`
-   :ref:`modindex`
-   :ref:`search`
```
Comment (HuangJunye) [inline docs/getting_started.rst]: These few lines appear in many places in the documentation and I never understand why they are there. Do you know why?

File: docs/getting_started/source.rst [hunk 3]
```diff
@@ -0,0 +1,419 @@
...
+   before you can install.  Most other Qiskit-family packages do not require
+   a compiler.
+
+Installing elements from source requires the following order of installation to
+prevent installing versions of elements that may be lower than those desired if the
+``pip`` version is behind the source versions:
+
+#. :ref:`qiskit-terra <install-qiskit-terra>`
+#. :ref:`qiskit-aer <install-qiskit-aer>`
+#. :ref:`qiskit-ignis <install-qiskit-ignis>`
```
Comment (HuangJunye) [inline docs/getting_started/source.rst]: Just a reminder that Ignis is getting deprecated and removed so we may need to change this part depending on the order of the PR getting merged. https://github.com/Qiskit/qiskit/pull/1487/

File: docs/maintainers_guide.rst [hunk 4]
```diff
@@ -0,0 +1,198 @@
+
+#################
+Maintainers Guide
+#################
+
+This document defines a *maintainer* as a contributor with merge privileges.
```
Comment (HuangJunye) [inline docs/maintainers_guide.rst:6]: Is there a link to the list of maintainers on GitHub?

Comment (HuangJunye) [reply docs/maintainers_guide.rst:6]: I think we should probably emphasize that this guide is probably not applicable to most users and direct them to the contributing to qiskit which is most relevant to most contributors.

**Review by HuangJunye — COMMENTED** [review]
I haven't had time to look at the new deprecation policy doc but this is looking great! I love the new contributing to qiskit and the flow chart. Great team work!

File: docs/getting_started.rst (hunk 2 above)
Comment (jakelishman) [reply docs/getting_started.rst]: The `.. Hiding - Indices and tables` starts with `..`, but will be failed to parse as a directive (it doesn't end in `::`, and isn't a hyperlink target), which means that reST considers this a comment.  The indented block following it is the rest of the "content" of the comment.

I imagine it's everywhere in the documentation just because people copy and paste.

File: docs/maintainers_guide.rst (hunk 4 above)
Comment (jakelishman) [reply docs/maintainers_guide.rst:6]: I think the first sentence is probably enough for that.  To get here, you already have to click on the "Maintainers' Guide" link, and the word "maintainer" already implies it's not for regular contributors, then the first sentence makes that clear.

File: docs/getting_started.rst (hunk 2 above)
Comment (HuangJunye) [reply docs/getting_started.rst]: I understand that it is a comment but I am wondering what these references and why they need to be hidden. Maybe like you said it was used once for some reasons and people just copy and paste.

Comment (jakelishman) [reply docs/getting_started.rst]: I think the default `index.rst` that Sphinx generates when you do `sphinx-quickstart` includes those lines, which is probably where they come from.  The actual text that would be rendered by them is the links to [the special names that Sphinx always generates](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#special-names).

File: docs/getting_started/local.rst (hunk 1 above)
Comment (jakelishman) [reply docs/getting_started/local.rst]: Yeah, this is a good way to just cut down some fluff.  Done in 3340ac8c8.

File: docs/deprecation_policy.rst [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+         "Qiskit Terra 0.20.0, and will be removed 3 months or more later. "
+         "Instead, you should use qiskit.other_function().",
+         category=DeprecationWarning,
+         stacklevel=2,
+      )
+      # ... the rest of the function ...
+
+Make sure you include the version of the package that introduced the deprecation
+warning (so maintainers can easily see when it is valid to remove it), and what
+the alternative path is.
```
Comment (HuangJunye) [inline docs/deprecation_policy.rst:145]: This is not directly related to this document itself but I can't help but think that it would be great if there is a package we can use to manage deprecation cycles. We have so many DeprecationWarnings everywhere, it's hard to keep track which ones need to be removed after what date. And I think if something is deprecated but not removed for a long time, users may start ignoring them.

File: docs/deprecation_policy.rst [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+
+   .. note::
+
+      These are *minimum* requirements.  For removal of core features, give
+      users an extra minor version.
+
+   *Reason*: there needs to be time for users to see these messages, and to give
+   them time to adjust.  Not all users will update their version of Qiskit
+   immediately, and some may skip minor versions.
+
```
Comment (HuangJunye) [inline docs/deprecation_policy.rst:79]: Should we add one more step for removing the feature completely together with the DeprecationWarnings? The tests for testing deprecations should also be removed right?

**Review by HuangJunye — COMMENTED** [review]
Reviewed the deprecation policy. It's very clear and concise! Just a few small comments.

File: docs/deprecation_policy.rst [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+   minor version immediately after the warnings.  Add an ``upgrade`` release
+   note that lists all the removals.  For example, if the alternative path was
+   provided in ``0.19.0`` and the warnings were added in ``0.20.0``, the
+   earliest version for removal is ``0.22.0``, even if ``0.21.0`` was released
+   more than three months after ``0.20.0``.
+
+   .. note::
+
+      These are *minimum* requirements.  For removal of core features, give
+      users an extra minor version.
```
Comment (mtreinish) [inline docs/deprecation_policy.rst]: For some functionality I don't think even one extra release is sufficient. We shouldn't set an upper bound on the length of deprecation, just the minimum amount of time we can guarantee something will keep working. I think wording this like we had before as just:

```suggestion
   .. note::

      These are *minimum* requirements.  For removal of significant or core features, give
      users at least an extra minor version if not longer.
```

File: docs/deprecation_policy.rst [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+#. The alternative path must be in place for one minor version before any
+   warnings are issued.  For example, if we want to replace the function ``foo()``
+   with ``bar()``, we must make at least one release with both functions before
+   issuing any warnings within ``foo()``.
+
+   *Reason*: we need to give people time to swap over without breaking their
+   code as soon as they upgrade.
+
+#. After one minor version, :ref:`issue the deprecation warnings
```
Comment (mtreinish) [inline docs/deprecation_policy.rst]: I don't understand "After one minor version" here. I assume you're talking about from the first point above, but not all deprecations are being replaced in a tick-tock pattern. I'd remove this beginning and just start with how to issue deprecation warnings, that or make this more general so it doesn't assume we're implementing the replacement immediately before deprecation.

File: docs/deprecation_policy.rst [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+Deprecation Policy
+##################
+
+Many users and other packages depend on different parts of Qiskit.  We need to
+make sure that whenever we make changes to the code, we give users ample time to
+adjust without breaking code that they have already written.
+
+Most importantly: don't *change* any interface that is public-facing unless we
+absolutely have to.  Adding things is ok, taking things away is discouraged
+but doable, while changing behavior is so difficult that it is best avoided.
```
Comment (mtreinish) [inline docs/deprecation_policy.rst]: I'd maybe word this a bit stronger. This document is our contract with our users about how and when we'd change things. But in general Qiskit is too eager to make breaking changes and we should try to push back against that here. 

The other thing I think worth calling out in a preamble is that we don't know how an API is necessarily being used or how widely it's being used. So we can't make assumptions around "nobody is using this" or "nobody depends on this behaviour"

File: docs/deprecation_policy.rst (hunk 5 above)
Comment (mtreinish) [reply docs/deprecation_policy.rst:145]: There are libraries to help with it, but honestly don't add too much value in my experience. Normally just decorates that have warning templates. The easiest way is to use decorators or annotations in the code which can be scanned.  

That being said, fundamentally I don't agree that having a long deprecation window is a bad thing, it's actually desirable so we give people a longer opportunity to migrate. Ideally I'd make the minimum waiting period at least a year. In other projects I typically restrict breaking changes to major versions and don't do them frequently at all (and I typically leave the N-1 major version open for bug fixes).

If a user starts ignoring a warning that's on them, if we follow the guide above they have a migration path and alternatives to migrate to which already have existed since prior to the warning. We're doing the best we can to signal that a change is coming and building the path for them to switch to the new pattern.

File: docs/deprecation_policy.rst [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+When removing a feature (for example a class, function or function parameter),
+we will follow this procedure:
+
+#. The alternative path must be in place for one minor version before any
+   warnings are issued.  For example, if we want to replace the function ``foo()``
+   with ``bar()``, we must make at least one release with both functions before
+   issuing any warnings within ``foo()``.
+
+   *Reason*: we need to give people time to swap over without breaking their
+   code as soon as they upgrade.
```
Comment (mtreinish) [inline docs/deprecation_policy.rst]: Did you want to mention `PendingDeprecationWarning` here. I like that we call it out in the old version, especially because it gives a good search string to find all the places to deprecate.

File: docs/deprecation_policy.rst [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+
+If you absolutely must change the behavior of existing code (other than fixing
+bugs), you will need to use your best judgment to apply the guiding principles
+at the top of this document.  The most appropriate warning for behavioral
+changes is usually ``FutureWarning``.  Some possibilities for how to effect a
+change:
+
+- If you are changing the default behavior of a function, consider adding a
+  keyword argument to select between old and new behaviors.  When it comes time,
+  you can issue a future warning if the keyword argument is not given
```
Comment (mtreinish) [inline docs/deprecation_policy.rst]:
```suggestion
  you can issue a ``FutureWarning`` if the keyword argument is not given
```

File: docs/deprecation_policy.rst [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+
+.. testing-deprecated-functionality:
+
+Testing deprecated functionality
+================================
+
+Whenever you add deprecation warnings, you will need to update tests involving
+the functionality.  The test suite should fail otherwise, because of the new
+warnings.
+
```
Comment (mtreinish) [inline docs/deprecation_policy.rst]: You should mention here that we need to continue testing the deprecated functionality to ensure it continues to work as is until the removal date.

File: docs/deprecation_policy.rst [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+   issuing any warnings within ``foo()``.
+
+   *Reason*: we need to give people time to swap over without breaking their
+   code as soon as they upgrade.
+
+#. After one minor version, :ref:`issue the deprecation warnings
+   <issuing-deprecation-warnings>`.  Add a release note with a ``deprecations``
+   section listing all deprecated paths, their alternatives, and the reason for
+   deprecation.  :ref:`Update the tests to test the warnings
+   <testing-deprecated-functionality>`.
```
Comment (mtreinish) [inline docs/deprecation_policy.rst]: We probably should explain more about what deprecation is here. Like the feature is frozen and now slated for removal but will continue to work as is with no changes (other than critical bug fixes) while in the deprecation window.

File: docs/deprecation_policy.rst [hunk 14]
```diff
@@ -0,0 +1,153 @@
...
+   *Reason*: removals must be highly visible for at least one version, to
+   minimize the surprise to users when they actually go.
+
+#. Set a removal date for the old feature.  This must be at least three months
+   after the version with the warnings was first released, and cannot be the
+   minor version immediately after the warnings.  Add an ``upgrade`` release
+   note that lists all the removals.  For example, if the alternative path was
+   provided in ``0.19.0`` and the warnings were added in ``0.20.0``, the
+   earliest version for removal is ``0.22.0``, even if ``0.21.0`` was released
+   more than three months after ``0.20.0``.
```
Comment (mtreinish) [inline docs/deprecation_policy.rst]: You probably should refer to the stable branch policy here and that a removal can only occur in a minor version release and not a patch release.

**Review by mtreinish — COMMENTED** [review]
Here are some initial comments. I started with just the deprecation policy section. I'll try to look at the other pieces later.

File: docs/maintainers_guide.rst [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+
+Adding New Tracked Elements
+---------------------------
+
+When a new Qiskit element is being added to the meta-package requirements, we
+need to increase the **Minor** version of the meta-package.
+
+For example, if the meta-package is tracking 2 elements ``qiskit-aer`` and
+``qiskit-terra`` and its version is ``0.7.4``. Then we release a new element
+``qiskit-ignis`` that we intend to also have included in the meta-package. When
```
Comment (y4izus) [inline docs/maintainers_guide.rst]: `qiskit-ignis`is deprecated, so I think we should remove every mention to it on the docs

File: docs/maintainers_guide.rst [hunk 16]
```diff
@@ -0,0 +1,198 @@
...
+Patch Version Increases
+-----------------------
+
+When any Qiskit element that is being already tracked by the meta-package
+releases a patch version to fix bugs in a release, we need also bump the
+requirement in the ``setup.py`` and then increase the patch version of the
+meta-package.
+
+For example, if the meta-package is tracking 3 elements ``qiskit-terra==0.8.1``,
+``qiskit-aer==0.2.1``, and ``qiskit-ignis==0.1.4`` with the current version
```
Comment (y4izus) [inline docs/maintainers_guide.rst]: Same

**Review by y4izus — COMMENTED** [review]
Great job!! thank you!! I've just added 2 comments, the rest LGTM 😄 👏

File: docs/maintainers_guide.rst (hunk 4 above)
Comment (1ucian0) [reply docs/maintainers_guide.rst:6]: > Is there a link to the list of maintainers on GitHub?

The closer things to that is https://github.com/orgs/Qiskit/people

File: docs/maintainers_guide.rst (hunk 15 above)
Comment (1ucian0) [reply docs/maintainers_guide.rst]: This is an examples. I think it can be changed in the following way to avoid confusion:

```suggestion
``qiskit-something`` that we intend to also have included in the meta-package. When
```

File: docs/maintainers_guide.rst (hunk 16 above)
Comment (1ucian0) [reply docs/maintainers_guide.rst]: Similar to https://github.com/Qiskit/qiskit/pull/1507#discussion_r887737685
```suggestion
``qiskit-aer==0.2.1``, and ``qiskit-something==0.1.4`` with the current version
```

File: docs/deprecation_policy.rst (hunk 14 above)
Comment (HuangJunye) [reply docs/deprecation_policy.rst]: Oh I was confused by the term `minor` release here (I thought it's `0.20.0` -> `0.20.1`) until you mention `patch` even though this is the standard semver language. Since we haven't reached 1.0 yet, maybe we should mention that `minor` release means changing the middle number e.g. `0.19.0` -> `0.20.0`.

File: docs/getting_started/source.rst (hunk 3 above)
Comment (1ucian0) [reply docs/getting_started/source.rst]: I think it boils down to the timeframe for this PR to be merged. Ignis is part of the metapackage until its next released, [planned for late June](https://github.com/qiskit-community/feedback/wiki/Release-Planning#qiskit-metapackage). If this PR merge gets too close (let's say, 1 week) to the metapackage release, then it makes sense remove every reference to ignis. Otherwise, I think it would be better to keep it and remove it later, even if the window for that is small.

I created https://github.com/1ucian0/qiskit/tree/rewrite-contributors-guide_wo_ignis on top of [`jakelishman:rewrite-contributors-guide`](https://github.com/jakelishman/qiskit/tree/rewrite-contributors-guide) to be ready quickly.

File: docs/deprecation_policy.rst [hunk 17]
```diff
@@ -0,0 +1,153 @@
...
+  deprecating the old function.
+
+- If you need to accept a new input that you cannot distinguish from an existing
+  possibility because of its type, consider letting it be passed by a different
+  keyword argument, or add a second function that only accepts the new form.
+
+
+.. issuing-deprecation-warnings:
+
+Issuing deprecation warnings
```
Comment (HuangJunye) [inline docs/deprecation_policy.rst:123]: I notice there is a section covering similar content on the qiskit-terra CONTRIBUTING.md. Should we remove that section and simply link to here? https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md#adding-deprecation-warnings

File: docs/deprecation_policy.rst (hunk 6 above)
Comment (jakelishman) [reply docs/deprecation_policy.rst:79]: This is part of the same point - I've just updated the wording to make that clearer.

File: docs/deprecation_policy.rst (hunk 8 above)
Comment (jakelishman) [reply docs/deprecation_policy.rst]: I've updated this to be clearer on both points - "After the alternative path has been in place for at least one minor version, ..." in 09069fc0a.

File: docs/deprecation_policy.rst (hunk 14 above)
Comment (jakelishman) [reply docs/deprecation_policy.rst]: @HuangJunye: I don't want to talk about that here, because the `major.minor.patch` terminology is standard semantic versioning, and it's the same terminology we use everywhere - if I go into too much detail here, it just becomes harder to read the policy at a glance.

File: docs/deprecation_policy.rst (hunk 17 above)
Comment (jakelishman) [reply docs/deprecation_policy.rst:123]: We can consider it after this merges, but it doesn't hurt to have an extra call out in the Terra contributing guide to point to this as well, I think.

File: docs/deprecation_policy.rst (hunk 9 above)
Comment (jakelishman) [reply docs/deprecation_policy.rst]: I've reworded and added your extra text in 09069fc0a.

File: docs/deprecation_policy.rst (hunk 10 above)
Comment (jakelishman) [reply docs/deprecation_policy.rst]: Done in 09069fc0a.

File: docs/deprecation_policy.rst (hunk 12 above)
Comment (jakelishman) [reply docs/deprecation_policy.rst]: Done in 09069fc0a.

File: docs/deprecation_policy.rst (hunk 13 above)
Comment (jakelishman) [reply docs/deprecation_policy.rst]: Done (in the paragraphs after the bullet points) in 09069fc0a.  I didn't want to make the list too long to keep it easy to scan / refer back to, but I expanded the text below it.

File: docs/deprecation_policy.rst (hunk 14 above)
Comment (jakelishman) [reply docs/deprecation_policy.rst]: I added text on the stable-branch policy to the preamble of the section, because I think it's relevant to every part of the policy, in 09069fc0a.

File: docs/maintainers_guide.rst (hunk 15 above)
Comment (jakelishman) [reply docs/maintainers_guide.rst]: Done in 03aca5c43.

File: docs/maintainers_guide.rst (hunk 16 above)
Comment (jakelishman) [reply docs/maintainers_guide.rst]: I just reduced this to two elements in 03aca5c43.

**Comment by jakelishman** [comment]
I've updated this with everyone's actionable comments, or (in I think only one case) commented as to why I haven't made the change as completely as requested.

I've left the reference to Ignis in the "building from source" guidelines, because Ignis isn't actually removed yet.  Luciano mentioned he's got a PR that fits on top of this one for when we do actually remove it.

File: docs/deprecation_policy.rst [hunk 18]
```diff
@@ -2,21 +2,36 @@
...
+absolutely have to.  Adding things is ok, taking things away is annoying for
+users but can be handled reasonably with plenty notice, but changing behavior
+generally means users cannot write code that will work with two subsequent
+versions of Qiskit, which is not acceptable.
+
+Beware that users will often be using functions, classes and methods that we,
+the Qiskit developers, may consider internal or not widely used.  Do not make
+assumptions that "this is buried, so nobody will be using it"; if it is public,
+it is subject to the policy.  The only exceptions here are functions and modules
+that are explicitly internal, *i.e.* those whose names begin with a leading
```
Comment (HuangJunye) [inline docs/deprecation_policy.rst:19]: It's good that here provide explicit definition of "internal functions". What about modules? Which modules are considered internal? Thinking about my PR to move fake backends from `qiskit.test.mock` to `qiskit.providers.fake_provider`. Are things inside `qiskit.test` considered internal? If so, are there other folders consider internal as well (for example, `qiskit.utils`)?

Comment (jakelishman) [reply docs/deprecation_policy.rst:19]: The text defines "explicitly internal" the same for both functions and modules - it's those with leading underscores, which is standard for Python.  The issue with `qiskit.test.mock` was that we (the developers) _kind of_ considered that internal because we didn't document it, but it didn't really have any explicit marking to it.  That's part of why we're moving it and documented it - people were using it, and we want to show that we're properly committing to that.

File: docs/deprecation_policy.rst [hunk 19]
```diff
@@ -29,27 +44,29 @@ we will follow this procedure:
...
 
-#. Set a removal date for the old feature.  This must be at least three months
-   after the version with the warnings was first released, and cannot be the
-   minor version immediately after the warnings.  Add an ``upgrade`` release
-   note that lists all the removals.  For example, if the alternative path was
-   provided in ``0.19.0`` and the warnings were added in ``0.20.0``, the
-   earliest version for removal is ``0.22.0``, even if ``0.21.0`` was released
-   more than three months after ``0.20.0``.
+#. Set a removal date for the old feature, and remove it (and the warnings) when
+   reached.  This must be at least three months after the version with the
```
Comment (HuangJunye) [inline docs/deprecation_policy.rst:63]: Do we set a specific removal date and remove it as soon as the date is reached? I thought we are going to just say it will be removed 3 months or later without providing a specific date.

In this comment, Matthew is suggesting to make the deprecation time even longer than 3 months. https://github.com/Qiskit/qiskit/pull/1507#discussion_r886799834

Comment (jakelishman) [reply docs/deprecation_policy.rst:63]: Yeah, the text says "at least three months" - that's the minimum.  Matt's comment mentions an ideal deprecation period of a year, which is something we'd _like_ to have in a fully mature project, but with the research status of Qiskit, it's not entirely achievable for everything right now.  We'll still try to give long deprecation periods, but I don't want to write "a year" into our contract with the users when we can't commit to it.

This text wasn't _entirely_ meant to mean we'll put a hard date in the actual messages, it was more for internal use.  That said, I do think that being clearer with users about when something will actually disappear is a good thing.

**Review by 1ucian0 — DISMISSED** [review]

File: docs/deprecation_policy.rst (hunk 19 above)
Comment (HuangJunye) [reply docs/deprecation_policy.rst:63]: > I do think that being clearer with users about when something will actually disappear is a good thing.

I think that would be great. What are the ways we can communicate that if not putting a hard date in the deprecation message? I also feel that even us the developers lose track of the various deprecated features and forget to remove them when they should be removed. 

In fact I am not even sure we know when a feature should be removed beyond the 3 months minimum time. Matthew's comment suggests to extend the deprecation period as long as possible until we absolutely need to remove it while from discussions with others I get the impressions that we want to minimize the number of features being deprecated and therefore should remove them as soon as possible. 

Sorry if the answers to these questions are very obvious due to my lack of software development experience.

Comment (jakelishman) [reply docs/deprecation_policy.rst:63]: Putting in a version number (rather than a date) for the removal is one way of doing that.  For remembering to effect these removals, we can try things like creating issues to remove and tagging them for particular milestones - that seemed to work ok for your work on `qiskit.test.mock`, for example.

We don't always know when the best time is, and it will be affected by use.  When we issue the warnings, we get user feedback (in places like the public Slack) from a wider audience about how that affects their workflows, and that gives us more information on when is best to go through with the removal.  We might delay that longer if the effects are larger than we anticipated (hopefully not), or stick to the contract if the backwards-compatibility support is holding us back from improving other things.

Comment (HuangJunye) [reply docs/deprecation_policy.rst:63]: I think creating an issue for removal is a good approach. We can then direct discussion of removal to that issue. Do you think this is something we should add to this guide?

**Review by HuangJunye — DISMISSED** [review]
LGTM!

**Comment by jakelishman** [comment]
I think Matthew wanted to look through this more / again, so let's hold off til then.  There's no particular urgency to get this merged - we can update the published documentation whenever we like, and as long as we merge it before the next metapackage minor release, it'll get archived into the next stable release.

**Comment by 1ucian0** [comment]
I merged in the removal of Ignis, as `qiskit 0.37` was released.

**Comment by jakelishman** [comment]
I've fixed the broken merge - it had restored a lot of the old contributing guidelines which this commit is intended to remove, and had missed the installation instructions for Ignis, since they'd moved.

File: docs/contributing_to_qiskit.rst [hunk 20]
```diff
@@ -7,689 +6,84 @@ Qiskit is an open-source project committed to bringing quantum computing to
...
-When you encounter a problem, please open an issue in the
-appropriate element's issue tracker:
+********************************
+Contributing to a Specific Repo
+********************************
 
+Each Qiskit package has its own set of Contributing Guidelines (kept in the ``CONTRIBUTING.md`` file) which
+details specific information on contributing to that repository. Make sure you read through the repo-specific
+Contributing Guidelines prior to making your contribution to a specific repo as each project may have
+slightly different requirements and processes.
```
Comment (mtreinish) [inline docs/contributing_to_qiskit.rst:43]: Heh, I think we have a circular loop, the top of all the per contributing guides say read the global contributing docs first :)

File: docs/contributing_to_qiskit.rst [hunk 21]
```diff
@@ -735,454 +129,97 @@ environments (e.g. one per supported Python version, for older versions of Qiski
...
+- All CI checks pass (it's recommended to run tests and lint checks locally before pushing).
+- New tests have for any new functionality that has been introduced.
+- The documentation has been updated accordingly for any new/modified functionality.
+- A release note has been added if the change has a user-facing impact.
+- Any superfluous comments or print statements have been removed.
+- All contributors have signed the :ref:`cla`.
+- The PR has a concise and explanatory title (e.g. ``Fixes Issue1234`` is a bad title!).
+- If the PR addresses an open issue the PR description includes the ``fixes #issue-number``
+  syntax to link the PR to that issue (**you must use the exact phrasing in order for GitHub
+  to automatically close the issue when the PR merges**)
```
Comment (mtreinish) [inline docs/contributing_to_qiskit.rst:182]: Would you want this to be a numbered list instead of bulleted list?

File: docs/contributing_to_qiskit.rst [hunk 22]
```diff
@@ -7,689 +6,84 @@ Qiskit is an open-source project committed to bringing quantum computing to
...
-   the commit message. Use the syntax::
-
-       Fixes #1234
-
-   if it fixes the issue (GitHub will close the issue when the PR merges).
-
-The main rule to follow is:
-
-The commit message must contain all the information required to fully
-understand and review the patch for correctness. Less is not more.
```
Comment (mtreinish) [inline docs/contributing_to_qiskit.rst]: Is this section preserved anywhere? I refer people to this section quite frequently when I'm asked about the contents of a good commit message.

File: docs/contributing_to_qiskit.rst [hunk 23]
```diff
@@ -7,689 +6,84 @@ Qiskit is an open-source project committed to bringing quantum computing to
...
-      we want to make sure that translated languages have enough community support
-      to build a translation team with translators, proofreaders, and translation leads.
-      If you want to be a translation lead or would be willing to join a new
-      translation project team, you can open a `GitHub issue <https://github.com/qiskit-community/qiskit-translations/issues/new/choose>`__
-      to start a discussion with the Qiskit team and recruit translation project members.
-#. Click the **Join** button and **paste the URL of your PR** in the dialog box where you
-   are asked why you want to join the Crowdin project.
-
-The administrators of the Crowdin project will review your request and give you
-access as quickly as they can.
```
Comment (mtreinish) [inline docs/contributing_to_qiskit.rst:599]: Similarly are we not preserving this documentation either? I've fielded many questions about how the documentation is structured and built and this guide is a good reference to point people to. Especially since it's not the most obvious system.

File: docs/getting_started.rst [hunk 24]
```diff
@@ -7,602 +7,12 @@ Getting started
...
-dependencies from source as part of the installation process. Support for these
-platforms are best effort only.
-
-Tier 3 platforms are currently:
-
- * Linux i686 (distributions compatible with the
-   `manylinux 2014 <https://www.python.org/dev/peps/pep-0599/>`__ packaging
-   specification) for Python >= 3.10
- * Windows 32 bit for Python >= 3.10
- * macOS arm64 (10.15 or newer)
```
Comment (mtreinish) [inline docs/getting_started.rst]: Are we dropping the support tiers description? I don't see it being added somewhere else, this is important that we document this somewhere, especially as we move to using more compiled extensions.

**Review by mtreinish — COMMENTED** [review]
Overall this LGTM, thanks for making the changes on the deprecation policy. I left a few inline comments. My only concerns right now are on losing some documentation that is still useful. We don't need to put it on the contributing guide necessarily but preserving those sections somewhere even if separate pages I think will be useful

File: docs/contributing_to_qiskit.rst (hunk 20 above)
Comment (1ucian0) [reply docs/contributing_to_qiskit.rst:43]: The repo-specific contributing guidelines says "read the general guidelines first". This one says rea the repo-specific guidelines before contribution. Would it help if this paragraph starts with "In addiition to this guildeline, ..." or "After reeding this guideline, remember that ..."?

Comment (mtreinish) [reply docs/contributing_to_qiskit.rst:43]: I was just joking here, I mean yeah if people read text like code it'd be an infinite loop but most people I think will be able to understand it. Maybe having a couple extra words here wouldn't hurt, but it's not critical.

File: docs/contributing_to_qiskit.rst (hunk 21 above)
Comment (javabster) [reply docs/contributing_to_qiskit.rst:182]: I think bullets are fine as people don't need to follow each step in order

File: docs/contributing_to_qiskit.rst (hunk 20 above)
Comment (javabster) [reply docs/contributing_to_qiskit.rst:43]: I think it's fine for them to link to each other and the more wordy we make it the less people will read 😅 . imho it's fine as is

File: docs/getting_started.rst (hunk 24 above)
Comment (javabster) [reply docs/getting_started.rst]: I personally don't think this is relevant for the contributing guidelines, and also I don't think we should use the guidelines as a dumping ground for documentation that we don't know where else to put 😅 . Do you have a suggestions for a more relevant place to keep this info? Perhaps the installation page?

File: docs/contributing_to_qiskit.rst (hunk 22 above)
Comment (javabster) [reply docs/contributing_to_qiskit.rst]: It's way too wordy and adds a lot of clutter to the page so we opted to cut it out. imho there are tons of resources available elsewhere on the internet that provides this kind of information that we can send people to.

File: docs/contributing_to_qiskit.rst (hunk 23 above)
Comment (javabster) [reply docs/contributing_to_qiskit.rst:599]: for translations this info is in the translations repo (which we link to higher up in the flowchart) so I think it can be removed from here (we generally went with the strategy of trying to reduce duplication as much as possible)

for the documentation build maybe this can go in the maintainer guide instead? It's a bit more advanced info and may change in future with the documentation updates so i'm hesitant to keep it in the doc we send first time contributors to

File: docs/getting_started.rst (hunk 24 above)
Comment (mtreinish) [reply docs/getting_started.rst]: It wasn't in the contributing guidelines, it was on the install/getting started page. This was deleted as part of splitting that into 3 separate pages instead of tabs. These are important details to document as part of the installation steps as platform support comes up quite often and it should be included on the local.rst file as part of that split.

Comment (javabster) [reply docs/getting_started.rst]: ahh sorry my bad i didn't look closely at the file name. I can add this section back in

**Review by mtreinish — APPROVED** [review]
This LGTM now, thanks for making the updates

**Comment by HuangJunye** [comment]
@javabster I am setting the development environment for my new laptop and I realized that a few parts of the original contributing to qiskit guide are missing after this PR is merged. Are they still somewhere in the latest doc? 

- https://qiskit.org/documentation/stable/0.36/contributing_to_qiskit.html#contributing-to-documentation
- https://qiskit.org/documentation/stable/0.36/contributing_to_qiskit.html#translating-documentation

**Comment by HuangJunye** [comment]
Sorry just found it in the review comment: https://github.com/Qiskit/qiskit/pull/1507#discussion_r923801173 I think the documentation structure and stuff can go into the maintainer's guide, but I think it would be good to have a section on how to contribute to documentation because it actually have a lower barrier of entry for contribution and we need a lot of help there. 

Related to https://github.com/Qiskit/qiskit-terra/issues/8274

**Comment by javabster** [comment]
hi @HuangJunye 

we discussed the documentation info and as it is currently written we thought it fits more into the maintainers guide, but if you would like to rewrite a more beginner-friendly section on how to contribute to documentation perhaps you could make a new PR to add that to the contributor guide?

As for the translation things we decided to cut that out as it seemed to just be a copy of what is in the translation repo so we preferred to link out to that. Then there is just a single source of truth for translations and we don't need to remember to keep 2 places updated

**Comment by HuangJunye** [comment]
Thanks for the reply :) I'll work on the section of how to contribute to documentation. 

For translations, I think it makes sense.
